### PR TITLE
Convert diff.groovy to java project

### LINKED ISSUE #105: diff-java-tool: convert diff.groovy to java project
convert diff.groovy to java project to manage it easily and be able to generate final JAR file and use it by github new workflow action. JAR file can be committed in main repo for simplicity.

### PR DESCRIPTION
**Pull request by relentless-pursuit**
Resolves #105

### CODE REVIEW

**Comment by romani** [comment]
Ok, as first step.

But where is isValidCheckstyleConfigsCombination ?
https://github.com/checkstyle/contribution/blob/41ec549dc9976dfcc919a5683b9e08a893e04d70/checkstyle-tester/diff.groovy#L123

We need whole code conversion.

**Comment by relentless-pursuit** [comment]
> Ok, as first step.
> 
> But where is isValidCheckstyleConfigsCombination ? https://github.com/checkstyle/contribution/blob/41ec549dc9976dfcc919a5683b9e08a893e04d70/checkstyle-tester/diff.groovy#L123
> 
> We need whole code conversion.

I have adde the method and some other missing method. To my understanding this is going to be a tough one. Because, I need to understand the sequence of groovy code and see how can i do it in java.

**Comment by romani** [comment]
Can we simply convert whole groovy to java ?
We need whole code, it will be required to make new workflow backwards compatible.

File: diff-java-tool/src/main/java/com/github/checkstyle/difftool/DiffTool.java [hunk 1]
```diff
@@ -0,0 +1,1115 @@
...
+                String sha = reader.readLine();
+                process.waitFor();
+                return sha != null ? sha.trim() : "";
+            }
+        }
+        throw new IllegalArgumentException("Error! Unknown " + repoType + " repository.");
+    }
+
+
+    // ... (other methods will follow)
```
Comment (romani) [inline diff-java-tool/src/main/java/com/github/checkstyle/difftool/DiffTool.java:909]: What is left?

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: diff-java-tool/src/main/java/com/github/checkstyle/difftool/DiffTool.java (hunk 1 above)
Comment (romani) [reply diff-java-tool/src/main/java/com/github/checkstyle/difftool/DiffTool.java:909]: As soon as we convert all, we can build jar, put it to releases and let checkstyle workflow simply download jar without any mess with cloning repo and building from scratch.

**Comment by romani** [comment]
@relentless-pursuit , ping

File: diff-java-tool/src/main/java/com/github/checkstyle/difftool/DiffTool.java (hunk 1 above)
Comment (relentless-pursuit) [reply diff-java-tool/src/main/java/com/github/checkstyle/difftool/DiffTool.java]: Done. Nothing, as you know i had to take help of gpt for conversion. So, it remained as a comment which i forgot to edit.

**Comment by relentless-pursuit** [comment]
The build is faling because some changes were done in main repo. Even though i ran the setup script and ran the mainslauncher in extractor, but i am pushing changes from folder diff-tools. For some reason it is not picking up those changes for me to commit.

**Review by romani — APPROVED** [review]
ok to merge,

testing will show where conversion was bad

**Pull request merged**